### PR TITLE
System stats page + refresh button

### PR DESCRIPTION
**Pull request by wilmol**
Solves #48 

Create new API endpoint **/api/stats/system** to get a few system stats

Create new option in menu modal:
palemoon_i7yejsteg2

Create new page **/stats/system**:
palemoon_ztvaoyf3nu

Rename old **/stats** page and API endpoint to **/stats/user**

Added a refresh button:
chrome_NgyQnPhHCY
Works on any page.
Useful for refreshing stat pages etc. without being logged out/losing session.

### CODE REVIEW

**Comment by wilmol** [comment]
Added a refresh button:
chrome_NgyQnPhHCY
Works on any page.
Useful for refreshing stat pages etc. without being logged out/losing session.

File: concord-server/server/src/main/java/com/rabidgremlin/concord/dao/SystemStatsDao.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  @SqlQuery("SELECT COUNT(*) " +
+      "FROM phrases p " +
+      "JOIN(SELECT phraseId FROM votes GROUP BY phraseId HAVING COUNT(phraseId) >= :consensusLevel) v " +
+      "ON p.phraseId = v.phraseId " +
+      "WHERE completed = false")
+  int getCountOfPhrasesWithConsensusThatAreNotCompleted(@Bind("consensusLevel") int consensusLevel);
+
+  @SqlQuery("SELECT COUNT(*) " +
+      "FROM (SELECT label FROM phrases WHERE label IS NOT NULL GROUP BY label) ALIAS")
```
Comment (joedai) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/dao/SystemStatsDao.java]: Change to:
> SELECT count(distinct label) FROM phrases WHERE label IS NOT NULL

File: concord-server/server/src/test/java/com/rabidgremlin/concord/resources/StatsResourceTest.java [hunk 2]
```diff
@@ -106,17 +114,44 @@ public void shouldFilterOutBulkUploadUser()
...
     assertThat(response.getEntity(), instanceOf(List.class));
-    assertEquals(0, ((List) response.getEntity()).size());
+    assertThat(((List) response.getEntity()).size(), is(0));
+  }
+
+  @Test
+  public void shouldGetSystemStats()
+  {
+    // Given
+    when(systemStatsDao.getTotalCountOfPhrases()).thenReturn(100);
```
Comment (joedai) [inline concord-server/server/src/test/java/com/rabidgremlin/concord/resources/StatsResourceTest.java]: For better readability, I'd assign all the numbers to local variables so you have a linkage between the input and out
i.e.
`int totoCountOfPhrases = 100`
`assertThat(response.getEntity(), is(new SystemStats(totoCountOfPhrases....`

File: concord-server/server/src/main/java/com/rabidgremlin/concord/api/SystemStats.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    return userCount;
+  }
+
+  public List<LabelCountStats> getLabelCountStats()
+  {
+    return labelCountStats;
+  }
+
+  @Override
+  public boolean equals(Object o)
```
Comment (LaurenceTews) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/api/SystemStats.java]: I can't see where you use this logic?

Comment (wilmol) [reply concord-server/server/src/main/java/com/rabidgremlin/concord/api/SystemStats.java]: in the unit tests, `assertThat` uses it under the hood

File: concord-ui/src/components/notfound.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+import '@rmwc/data-table/data-table.css';
+
+export class NotFound extends Component {
+  render() {
+    return (
+      <div>
+        <p>404 Not Found</p>
```
Comment (LaurenceTews) [inline concord-ui/src/components/notfound.js]: Surely a bit of styling? Also would be nice to add a 'return to home' button here.

File: concord-ui/src/components/notfound.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+import '@rmwc/data-table/data-table.css';
```
Comment (LaurenceTews) [inline concord-ui/src/components/notfound.js]: unused?

File: concord-ui/src/components/systemstatstable.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  componentDidMount() {
+    this.props.dispatch(getSystemStats());
+  }
+
+  componentDidUpdate(oldProps) {
+    const newProps = this.props;
+    if (oldProps.loading !== newProps.loading && newProps.statsData) {
+      const data = newProps.statsData;
```
Comment (LaurenceTews) [inline concord-ui/src/components/systemstatstable.js]: Should be able to do some destructuring here. e.g.

const { totalPhrases, completedPhrases, phrasesWithConsensus } = newProps.statsData;

Then you won't have to keep referencing the parent node.

**Comment by wilmol** [comment]
Going to turn this into an 'Admin' type page

**Pull request closed without merging**